Created blogpost for Observability Game days at Skyscanner (#4039)
open-telemetry/opentelemetry.io: content/en/blog/2024/demo-skyscanner/index.md

---
title:
  "Making observability fun: How we increased engineers' confidence in incident
  management using a game"
linkTitle: Skyscanner using OTel Demo

> date: 2024-02-22

Reviewer: ```suggestion
date: 2024-02-26
```

---
title:

>   'Making observability fun: How we increased engineers confidence in incident

Reviewer: Observability is always fun, no need to make it fun 😢 ... just kidding! great title :-)

>   management using a game'

Reviewer 2: Looking at how blog titles are displayed on the side bar, would this title be potentially too long?

![Screenshot 2024-02-22 at 14 45 42](https://github.com/open-telemetry/opentelemetry.io/assets/4158734/f1b133ed-58b6-4a71-8d91-b686599e7aa0)

> linkTitle: Observability game day

Reviewer 2: I presume this is the title that appears on the side bar. Should it mention Skyscanner or the OTel Demo?

Author: I have updated the linkTitle :)

> linkTitle: Skyscanner using OtelDemo

Reviewer 2: (nit) I think it's normally spelt OTel Demo :)

Reviewer: ```suggestion
linkTitle: Skyscanner using OTel Demo
```

> date: 2024-02-22

Reviewer: ```suggestion
date: 2024-02-22
```
Keep this suggestion open until we figured out the publication date

author: >-
  [Jordi Bisbal Ansaldo](https://github.com/jordibisbal8) (Skyscanner)
---

> At Skyscanner, as in many organisations, teams tend to follow specific runbooks

Reviewer: ```suggestion
At [Skyscanner](https://www.skyscanner.com), as in many organisations, teams tend to follow specific runbooks
```

practically useless.

After migrating our telemetry data to the OpenTelemetry standards, at
Skyscanner, we now have richer instrumentation and can rely on observability
directly. As a result, we are ready to adopt a new
[observability mindset](https://charity.wtf/2019/09/20/love-and-alerting-in-the-time-of-cholera-and-observability/),
which requires training our engineers to work effectively with the new
ecosystem. This allows them to react efficiently to any known or unknown issues,
even under pressure.

To achieve this, we believe that the best way to gain knowledge isn’t through
one-time viewings of documents or videos. Instead, it’s through practical
exercises that include situations with never-before-seen (or at least rarely
seen) problems. This helps the company reduce the time to mitigate an issue
(TTM), which goes from when a first responder acknowledges the incident to the
time users stop suffering from the incident.

## Environment

To begin with, we need to set up an environment that demonstrates the best
practices for monitoring and debugging using OpenTelemetry instrumentation and
observability. For this, we propose the use of the official

> [OpenTelemetry Demo project](https://github.com/open-telemetry/opentelemetry-demo),

Reviewer: ```suggestion
[OpenTelemetry Demo](/docs/demo/),
```

which is a realistic example of a distributed system called Astronomy Shop.

> Thanks to its OpenTelemetry Protocol (OTLP), it allows us to simply point the

Reviewer: ```suggestion
Thanks to the [OpenTelemetry Protocol](/docs/specs/otlp/) (OTLP), it allows us to simply point the
```

standard OTLP exporter in the Collector to [New Relic](https://newrelic.com/),
our chosen observability platform at Skyscanner which, like other platforms, is

> fully embracing open standards to ingest telemetry data.

Reviewer: @open-telemetry/docs-approvers double checking if we are OK with naming an observability vendor or if this should be "to our chosen observability platform". I am OK with it since it's an end-user blog post.

Reviewer 2: This is still a valid point (although outdated). Considering the trace screenshot below comes from New Relic, it makes me think it's okay to mention it here. I don't think the blogpost is trying to advertise New Relic features except OTLP ingest, and in any case it compares it to other platforms that also accept OTLP.

This system contains regressions that can be injected into the platform and
helps us demonstrate the importance of Service Levels Objectives (SLOs),
tracing, logs, metrics, etc. For instance, we can observe traffic flow through
various components, as shown in the image below. Since part of the OpenTelemetry

> ecosystem is open-source, we can easily introduce any new features that will be

Reviewer: ```suggestion
ecosystem is open source, we can easily introduce any new features that will be
```

reviewed by OpenTelemetry contributors.

![Distributed tracing example in Astronomy shop](tracing_example.png)

## Observability game day

Once the environment is set up, we can introduce the Observability Game Day, an
initiative based on the Wheel of Misfortune practices that Google uses and
describes in the [Site Reliability Engineering book](https://sre.google/books/).

This game simulates a production incident, where a moderator known as the game
master (GM) conducts the session and someone from the audience spins the wheel
and explains an incident or outage. The participants are then divided into teams
and tasked with identifying and resolving the issue as quickly as possible. If
the solution is not optimal, the GM can help by introducing a new tool or view,
which gives a different perspective on how to tackle the incident (knowledge
sharing). This exercise can be repeated multiple times for different incidents.

> ![Wheel of misfortune example](wheel.png)

Reviewer: did you create this image or is it taken out of the book?

Author: IIRC, I created it using an online tool where I was able to add the desired text :)

Reviewer: ok, that's all I wanted to know

## Results

The Observability Game Day has already been completed by multiple Skyscanner
teams, where each team observability expert (ambassador) runs the session. The
participants have given extremely positive feedback, where 90% of the responders
say that after the Game Day, they feel more confident debugging production
systems and would love to have further sessions.

- Hugely valuable to run against real services and to compare and contrast
  different debugging methods. I'm certain everyone, regardless of skill level,
  will have got something out of the session - I know I did! Thank you for

>   taking the time to set this up and promoting it for us - Dominic Fraser
>   (Senior Software Engineer)

Reviewer: ```suggestion
  taking the time to set this up and promoting it for us - Dominic Fraser
  (Senior Software Engineer)
```

Are those people OK with being named here, and if so, we can link their github accounts

Author: Yes, they are more than fine :) One of them has and the other not, should we still link the one?

- It is a really great (company-wide) initiative to get people upskilled in
  observability and OpenTelemetry/New Relic and I personally found it very

>   useful, as well as a lot of fun! :D” - Polly Yankova (Software Engineer)

Reviewer: ```suggestion
  useful, as well as a lot of fun! :D - Polly Yankova (Software Engineer)
```

## Next steps

The next action is to run sessions for the entire company and convert them into
a Skyscanner learning course. This way, the content can be used during the
onboarding process for new joiners or even reviewed at any time as a refresher
for those who have been in the company longer. In addition, after observing
common feedback, we identified that it would be beneficial to extend the current
incidents to include more front-end-specific ones, such as incidents triggered

> by browser traffic. To achieve this, we have contributed to the OpenTelemetry
> Demo project and enabled these features for other interested parties. For more

Reviewer: ```suggestion
by browser traffic. To achieve this, we have contributed to the OpenTelemetry
Demo and enabled these features for other interested parties. For more
```

> information, please have a look at the raised
> [PR](https://github.com/open-telemetry/opentelemetry-demo/pull/1345).

Reviewer: ```suggestion
information, please have a look at the
[raised PR](https://github.com/open-telemetry/opentelemetry-demo/pull/1345).
```

makes the link a little bit more visibile.
